src/highdicom/seg/sop.py

```diff
@@ -342,6 +358,49 @@ def __init__(
             **kwargs
         )
 
+        # General Reference
+        self.SourceImageSequence: List[Dataset] = []
+        referenced_series: Dict[str, List[Dataset]] = defaultdict(list)
+        for img in source_images:
+            if is_multiframe:
```

Comment:
This is actually more restrictive than the current master. It is fine for the number of segmentation planes not to match the number of source planes if the plane positions have been explicitly specified (and the number of those matches the segmentation array, but this is checked later). You may want to also update the exception message in line with this

```suggestion
            if is_multiframe and plane_positions is None:
```